Orange/widgets/visualize/utils/plotgui.py

```diff
@@ -0,0 +1,113 @@
+from AnyQt.QtWidgets import QAction
+from AnyQt.QtCore import Qt
+from AnyQt.QtGui import QKeySequence
+
+from Orange.widgets import gui
+
+
+JITTER_SIZES = [0, 0.1, 0.5, 1, 2, 3, 4, 5, 7, 10]
+
+
+def jittering(master, parent, jitter_sizes=None, is_cb_jitter_cont=False):
+    jitter_sizes = jitter_sizes if isinstance(jitter_sizes, list) else JITTER_SIZES
+    gui.valueSlider(
+        widget=parent, master=master, value='graph.jitter_size',
+        label="Jittering: ", values=jitter_sizes, callback=master.reset_graph_data,
+        labelFormat=lambda x:
+        "None" if x == 0 else ("%.1f %%" if x < 1 else "%d %%") % x)
+    if is_cb_jitter_cont:
+        gui.checkBox(
+            widget=gui.indentedBox(widget=parent),
+            master=master, value="graph.jitter_continuous",
+            label="Jitter numeric values", callback=master.reset_graph_data)
+
+def box_axis_data(master, parent):
+    common_options = dict(
+        labelWidth=50, orientation=Qt.Horizontal, sendSelectedValue=True,
+        valueType=str)
+    box_axis_data = gui.vBox(parent, "Axis Data")
+    master.cb_attr_x = gui.comboBox(
+        box_axis_data, master, "attr_x", label="Axis x:", callback=master.update_attr,
+        model=master.xy_model, **common_options)
+    master.cb_attr_y = gui.comboBox(
+        box_axis_data, master, "attr_y", label="Axis y:", callback=master.update_attr,
+        model=master.xy_model, **common_options)
+    return box_axis_data
```

Comment:
Isn't this specific to scatter plot? It can stay in the scatter plot's `__init__`.

Reply:
Yes, probably.